SYCL/ESIMD/api/functional/type_coverage.hpp

```diff
@@ -234,7 +234,7 @@ inline bool for_all_combinations() {
 //  fp_extra - specific, non C++ data types
 //  uint - all unsigned C++ integral data types
 //  sint - all signed C++ integral data types
-enum class tested_types { core, fp, fp_extra, uint, sint };
+enum class tested_types { core, small, fp, fp_extra, uint, sint };
```

Comment:
The `small` doesn't look like something related to the `core`. Maybe something like this?
```suggestion
enum class tested_types { core, core_ci, fp, fp_extra, uint, sint };
```
or this
```suggestion
enum class tested_types { core_all, core, fp, fp_extra, uint, sint };
```

Please, don't forget to update related comments for every code update.

Reply:
On the second thought, why not to use `FULL_TYPE_COVERAGE` directly within `get_tested_types()`?
By doing so:

- there would be no strict requirement to use `FULL_TYPE_COVERAGE` within every `cpp` (only if needed)
- we would be able to have lesser `tested_types::uint` and `tested_types::sint` sets

Reply:
> On the second thought, why not to use FULL_TYPE_COVERAGE directly within get_tested_types()?

Yes, we ca. but by current solution we can explicitly see the limitations in each test

Reply:
Right, but we would need `uint_small`, `sint_small` with current implementation. And maybe `fp_extra_small` or even `fp_small` in future. Feel free to suggest a different solution for this concern though.

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/832/commits/e162bda82d57be4839dc67b58c109b6f7cf9b438#diff-3bbc1ab1445c827282cc08d73f76b58cf553d9ced2c799ba46e6148583896d22R243-R255)